bundles/core/org.eclipse.smarthome.core.library/src/main/java/org/eclipse/smarthome/core/library/types/PointType.java

```diff
@@ -0,0 +1,163 @@
+    private static final BigDecimal flat = new BigDecimal(180);
+    private static final BigDecimal right = new BigDecimal(90);
+
+    public PointType(DecimalType latitude, DecimalType longitude) {
+        canonicalize(latitude, longitude);
+    }
+
+    public PointType(DecimalType latitude, DecimalType longitude, DecimalType altitude) {
+        this(latitude, longitude);
+        setAltitude(altitude);
+    }
+
+    public PointType(StringType latitude, StringType longitude) {
+        this(new DecimalType(latitude.toString()), new DecimalType(longitude.toString()));
+    }
+
+    public PointType(StringType latitude, StringType longitude, StringType altitude) {
+        this(new DecimalType(latitude.toString()), new DecimalType(longitude.toString()), new DecimalType(
+                altitude.toString()));
+    }
+
+    public PointType(String value) {
+        if (!value.isEmpty()) {
+            String[] elements = value.split(",");
+            if (elements.length >= 2) {
+                canonicalize(new DecimalType(elements[0]), new DecimalType(elements[1]));
+                if (elements.length == 3) {
+                    setAltitude(new DecimalType(elements[2]));
+                }
+            }
+        }
+    }
+
+    public DecimalType getLatitude() {
+        return new DecimalType(latitude);
+    }
+
+    public DecimalType getLongitude() {
+        return new DecimalType(longitude);
+    }
+
+    public DecimalType getAltitude() {
+        return new DecimalType(altitude);
+    }
+
+    public void setAltitude(DecimalType altitude) {
+        this.altitude = altitude.toBigDecimal();
+    }
+
+    public DecimalType getGravity() {
+        double latRad = Math.toRadians(latitude.doubleValue());
+        double deltaG = -2000.0 * (altitude.doubleValue() / 1000) * EARTH_GRAVITATIONAL_CONSTANT
+                / (Math.pow(WGS84_a, 3.0));
+        double sin2lat = Math.sin(latRad) * Math.sin(latRad);
+        double sin22lat = Math.sin(2.0 * latRad) * Math.sin(2.0 * latRad);
+        double result = (9.780327 * (1.0 + 5.3024e-3 * sin2lat - 5.8e-6 * sin22lat) + deltaG);
+        return new DecimalType(result);
+    }
+
+    /**
+     * <p>
+     * Formats the value of this type according to a pattern (@see {@link Formatter}). One single value of this type can
+     * be referenced by the pattern using an index. The item order is defined by the natural (alphabetical) order of
+     * their keys.
+     * </p>
+     *
+     * @param pattern the pattern to use containing indexes to reference the
+     *            single elements of this type.
+     */
+    @Override
+    public String format(String pattern) {
+        return String.format(pattern, getConstituents().values().toArray());
+    }
+
+    public PointType valueOf(String value) {
+        return new PointType(value);
+    }
+
+    @Override
+    public String toString() {
+        return String.format("%1$.2f°N, %2$.2f°W, %2$.2f m", latitude, longitude, altitude);
```

Comment:
@clinique: I just noticed that there is a bug: A type should be able to parse its own toString() representation in the valueOf() method, but this fails since valueOf expects comma-separated decimals without any further characters.
I would actually suggest to change the toString to do it this way - this would be e.g. in line with the HSBType. The rest is rather formatting, which should happen in a state description for the UI only. Would you agree with this change? Would need to be changed in openHAB 1 as well I guess...